### PR TITLE
LPS-164938 "Select File" button is displayed and clickable on My Workflow Tasks

### PR DESCRIPTION
**Pull request by Manda-vit**
Related issue: https://issues.liferay.com/browse/LPS-164938

### CODE REVIEW

**Comment by Manda-vit** [comment]
ci:test:relevant

**Comment by Manda-vit** [comment]
ci:test:relevant

**Comment by Manda-vit** [comment]
ci:test:relevant

File: modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/java/com/liferay/object/dynamic/data/mapping/form/field/type/internal/attachment/AttachmentDDMFormFieldTemplateContextContributor.java [hunk 1]
```diff
@@ -84,13 +84,19 @@ public Map<String, Object> getParameters(
...
+				}
+
+				return _language.format(
+					ddmFormFieldRenderingContext.getLocale(),
+					"upload-a-x-no-larger-than-x-mb",
+					new Object[] {
+						ddmFormField.getProperty("acceptedFileExtensions"),
+						maximumFileSize
+					});
+			}
```
Comment (danvb112) [inline modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/java/com/liferay/object/dynamic/data/mapping/form/field/type/internal/attachment/AttachmentDDMFormFieldTemplateContextContributor.java:99]: Hey @Manda-vit, I don't think we need to do this validation on the backend, if you see that this `tip` property is being passed to the frontend, then I think we'd better get it on the frontend and do the validation there, dropping this backend validation.

1. For that you can add a new `tip` prop of type `string` in `IProps` to `Attachment.tsx` and in the Attachment component itself you can already have access to this property in the frontend.

2. The component responsible for showing this `tip` message is `FieldBase`, so we make the condition in it, like:

```tsx
<FieldBase
	readOnly={readOnly}
	tip={!readOnly ? tip : ""} 
	{...otherProps} 
	{...error}
>
```

File: modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Attachment/Attachment.tsx [hunk 2]
```diff
@@ -237,10 +243,11 @@ export default function Attachment({
 	};
 
 	return (
-		<FieldBase {...otherProps} {...error}>
+		<FieldBase {...otherProps} {...error} readOnly={readOnly}>
```
Comment (danvb112) [inline modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Attachment/Attachment.tsx:246]: Hey @Manda-vit, Just to let you know that we use a pattern here to always pass the rest props as the last of the component, like:

```tsx
<FieldBase readOnly={readOnly} {...otherProps} {...error}>
```

File: modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Attachment/Attachment.tsx [hunk 3]
```diff
@@ -237,10 +243,11 @@ export default function Attachment({
...
 			<div className="inline-item lfr-objects__attachment">
 				<ClayButton
 					className="lfr-objects__attachment-button"
+					disabled={readOnly}
 					displayType="secondary"
 					onClick={() => {
 						setError({});
```
Comment (danvb112) [inline modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Attachment/Attachment.tsx:253]: Hey @Manda-vit, from what I saw in the ticket I think that this button should not appear on the screen when `readOnly` is `true`, so we can make a condition for it very similar to what you did with the delete and download button, and you can remove the `disabled` property from the button itself:

```tsx
{!readOnly && (
	<ClayButton
		className="lfr-objects__attachment-button"
		displayType="secondary"
		onClick={() => {
			setError({});

			if (fileSource === 'documentsAndMedia') {
				openSelectionModal({
					onSelect: handleSelectedItem,
					selectEventName: `${portletNamespace}selectAttachmentEntry`,
					title: Liferay.Language.get('select-file'),
					url,
				});
			}
			else if (fileSource === 'userComputer') {
				const filePicker = inputRef.current;
				if (filePicker) {
					filePicker.value = '';
					filePicker.click();
				}
			}
		}}
	>
		{Liferay.Language.get('select-file')}
	</ClayButton>
)}
```

**Review by danvb112 — CHANGES_REQUESTED** [review]
Hey @Manda-vit Great work on this bug, I just have a few suggestions, if you have any questions just ask :smile:

**Comment by Manda-vit** [comment]
ci:test:sf

**Review by danvb112 — APPROVED** [review]
LGTM! :tada:

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by Manda-vit** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by Manda-vit** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:forward

**Pull request closed without merging**